### ISSUE TITLE
[Bug]: HeaderNavigation causes hydration to fail on mobile

### ISSUE THREAD

**Issue by mdavo6** [vuestorefront/spree#241]
### Expected Behavior

Page loads correctly with no errors.

### Actual Behavior

Console raises the following error:
```
The client-side rendered virtual DOM tree is not matching server-rendered content. This is likely caused by incorrect HTML markup, for example nesting block-level elements inside <p>, or missing <tbody>. Bailing hydration and performing full client-side render.
```

### Possible Solution

I have narrowed down the error to the following line:
```
<div class="sf-header__navigation desktop" v-if="!isMobile">

</div>
```
SOLUTION PROVIDED IN NEW COMMENT BELOW

### Steps to reproduce

- Load homepage on fresh install of Spree Vue.
- Open devtools and switch to view on mobile device.
- Trigger a full page refresh.
- View console. Error should now be present.

### What version of Spree integration are you using?

Spree 4.3

### What version of Node.js are you using?

14.19.3

### What browser (and version) are you using?

Chrome

### What operating system (and version) are you using?

Ubuntu

### Relevant log output

```shell
[Vue warn]: The client-side rendered virtual DOM tree is not matching server-rendered content. This is likely caused by incorrect HTML markup, for example nesting block-level elements inside <p>, or missing <tbody>. Bailing hydration and performing full client-side render.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by mdavo6** [comment]
May be related to this: https://github.com/vuestorefront/storefront-ui/issues/2019

**Comment by rafalcymerys** [comment]
Good catch @mdavo6 , thank you for bringing this up.
I'd need to double check that, but if I remember correctly, the latest versions of storefront-ui the mobileObserver (the utility that provides the value for isMobile) was removed. 
The solution you proposed makes total sense. Could you send a PR? :)

**Comment by mdavo6** [comment]
Not a problem @rafalcymerys  Glad to hear it makes sense. I will prepare a PR now.

**Comment by rafalcymerys** [comment]
Thanks!

**Comment by mdavo6** [comment]
Closed per #242
